railties/test/generators/app_generator_test.rb

```diff
@@ -886,10 +848,10 @@ def test_skip_hotwire
 
   def test_css_option_with_asset_pipeline_tailwind
     run_generator_and_bundler [destination_root, "--css=tailwind"]
+
+
+    assert_includes @rails_commands, "tailwindcss:install", "`tailwindcss:install` expected to be called, but wasn't."
```

Comment:
Same comment here regarding test coverage in the `tailwindcss-rails` gem and changing `run_generator_and_bundler` to `run_generator`.

Reply:
I'm currently working on PRs to add those tests. :smiley:  I started with https://github.com/rails/importmap-rails/pull/206 because `AppGeneratorTest` actually doesn't cover much of the behavior of `importmap:install`.  I also just submitted https://github.com/rails/dartsass-rails/pull/49, but I don't have commit privileges for that repo.

As those PRs are completed I will follow up with Rails PRs to update `AppGeneratorTest`.  And I will be sure to add you as a co-author, @zzak.

@flavorjones Do you think `tailwindcss-rails` would benefit from the approach I took in https://github.com/rails/dartsass-rails/pull/49?  Specifically, the way it tests against freshly generated Rails apps from multiple versions of Rails, including Rails `main` branch?

Reply:
@jonathanhefner If you mean "using appraisal to matrix test against rails", sure, that would be useful, I think. (There is existing coverage, run against Rails 7.1 and edge, for most of the gem (and a reminder that my involvement in the development of the gem has been minimal, I'm mostly a release manager at this point).)

But I also think the risk of breakage is relatively low, so I don't strongly feel like it needs to be added _right now_.

It seems likely that we'll end up adding some kind of support for engines soon, and I'll want to add matrix coverage at that point for sure. So I'm happy to defer more thorough coverage for a bit.

Reply:
I've opened hotwired/turbo-rails#553 and hotwired/stimulus-rails#136 based on Jonathan's work here.

Looking into adding tests to jsbundling-rails and cssbundling-rails, but they currently have no testing so might take a little longer to setup.

Reply:
@zzak Thank you! :smile:  I submitted https://github.com/rails/jsbundling-rails/pull/182 the other day, and I am working on similar tests for `cssbundling-rails`.

Reply:
Awesome, thanks for following up!! I will resolve this thread since I think we will have PRs open (or merged) for all of the upstream libraries properly.